examples/advent2024/BUILD

```
@@ -9,50 +9,14 @@ utils = [
     "sort.carbon",
 ]
 
-carbon_binary(
-    name = "day1_part1",
-    srcs = [
-        "day1_common.carbon",
-        "day1_part1.carbon",
-    ] + utils,
-)
-
-carbon_binary(
-    name = "day1_part2",
-    srcs = [
-        "day1_common.carbon",
-        "day1_part2.carbon",
-    ] + utils,
-)
-
-carbon_binary(
-    name = "day2_part1",
-    srcs = [
-        "day2_common.carbon",
-        "day2_part1.carbon",
-    ] + utils,
-)
-
-carbon_binary(
-    name = "day2_part2",
-    srcs = [
-        "day2_common.carbon",
-        "day2_part2.carbon",
-    ] + utils,
-)
-
-carbon_binary(
-    name = "day3_part1",
-    srcs = [
-        "day3_common.carbon",
-        "day3_part1.carbon",
-    ] + utils,
-)
-
-carbon_binary(
-    name = "day3_part2",
-    srcs = [
-        "day3_common.carbon",
-        "day3_part2.carbon",
-    ] + utils,
-)
+[
+    carbon_binary(
+        name = "day%d_part%d" % (day, part),
+        srcs = [
+            "day%d_common.carbon" % day,
+            "day%d_part%d.carbon" % (day, part),
```

Had you considered using \.format instead? I note this because I think we're predominantly using it in `.bzl` files (somehow I think this may be the first case in a BUILD)

```suggestion
        name = "day{0}_part{1}".format(day, part),
        srcs = [
            "day{0}_common.carbon".format(day),
            "day{0}_part{1}.carbon".format(day, part),
```